### PR TITLE
access control and cors fix

### PR DESCRIPTION
**Pull request by bensenescu**

### CODE REVIEW

File: src/components/AccessControl/AccessControl.vue [hunk 1]
```diff
@@ -5,19 +5,35 @@
...
+        case 2:
+          return 'admin';
+        default:
+          return 'error';
+      }
+    },
     hasRole() {
-      return (this.roles.length === 0 || this.roles.includes(this.role))
-          || (this.componentUserId === this.userId);
+      return (this.roles.length === 0 || this.roles.includes(this.userRole));
```
Comment (ZachOBrien) [inline src/components/AccessControl/AccessControl.vue:36]: Does this just mean we're ditching the idea of a user having control over something they created?

File: src/components/AccessControl/AccessControl.vue [hunk 2]
```diff
@@ -5,19 +5,35 @@
...
+    userRole() {
+      switch (this.userRoleNum) {
+        case 0:
+          return 'user';
+        case 2:
+          return 'admin';
+        default:
+          return 'error';
+      }
+    },
```
Comment (ZachOBrien) [inline src/components/AccessControl/AccessControl.vue:34]: Nice this is all really cool

File: src/components/AccessControl/AccessControl.vue (hunk 1 above)
Comment (bensenescu) [reply src/components/AccessControl/AccessControl.vue:36]: Only admins can create things though right?

File: src/utils/auth/state/userState.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ class UserState {
     if (this.getIsSignedIn()) {
       const payload = jwtUtils.getTokenPayload(tokenService.getAccessToken());
       if (payload !== null) {
-        return payload.adminLevel;
+        return payload.privilegeLevel;
```
Comment (jackblanc) [inline src/utils/auth/state/userState.js:50]: This broke a test, what is it doing?

File: src/components/AccessControl/AccessControl.vue [hunk 4]
```diff
@@ -5,19 +5,35 @@
...
+    };
+  },
+  created() {
+    this.userRoleNum = userState.getUserAdminLevel();
+  },
   computed: {
+    userRole() {
+      switch (this.userRoleNum) {
+        case 0:
+          return 'user';
```
Comment (jackblanc) [inline src/components/AccessControl/AccessControl.vue:28]: Can we use a map of code to user here with objects?

Comment (jackblanc) [reply src/components/AccessControl/AccessControl.vue:28]: There will be other places we reference these values.

File: src/utils/auth/state/userState.js (hunk 3 above)
Comment (bensenescu) [reply src/utils/auth/state/userState.js:50]: adminlevel is not a field in payload. PrivilidgeLevel is what its called now. Ex. user = 0, admin = 2. The test most likely needs to be updated. my bbbb

File: src/components/AccessControl/AccessControl.vue (hunk 4 above)
Comment (bensenescu) [reply src/components/AccessControl/AccessControl.vue:28]: If we need to use that code else where in the frontend, we should use a map. However, the idea of this component (thanks @ZachOBrien ) is that you don't need to know what privilege level users are anywhere else.

**Pull request merged**